### ISSUE TITLE
Keptn installation on AKS failing

### ISSUE THREAD

**Issue by jetzlstorfer** [keptn/keptn#1429]
When installing Keptn on AKS the installation is failing, since pods are not coming up before the installation times out:
- `configuration-service` pod in keptn namespace is stuck in the container-creating state

**Environment**

* Client OS (e.g., Linux, OSX, Windows): mac os 10.14.6
* Keptn Version (`keptn version`): keptn 0.6.0
* Kubernetes Cloud Provider (e.g., GKE, AKS): AKS, cluster version 1.14.8, tested with several node sizes, last test was on cluster with 2x b4ms nodes
* Kubernetes version (`kubectl version`): 1.14.7

**Affected Component**

* [ ] Docs
* [x] CLI
* [x] Kubernetes Integration
* [ ] Openshift Integration
* [ ] Helm
* [ ] Istio
* [ ] Bridge
* [ ] Gatekeeper
* [ ] Datastore
* [ ] REST API
* [ ] eventbroker / distributor
* [ ] jmeter

**Describe the bug**
Keptn installation fails when waiting for all pods in the keptn namespace and keptn-datastore namespace to come up.

**To Reproduce**
1. Create cluster in AKS, e.g., a cluster with 2 nodes each a b4ms instance
1. Wait for the cluster to be created
1. Install keptn with `keptn install --platform=aks --use-case=quality-gates` --verbose
1. Wait for the installation to fail since the pods `configuration-service` in the namespace keptn is not coming up before the timeout.

**Expected behavior**
The installation should run through without failing.

**Current behavior**
The installation is failing.

**Log output**
```
[keptn|DEBUG] [2020-02-24 19:38:18] Retry: 18/24 - Wait 10s for pods to start in namespace keptn ...
NAME                                                           READY   STATUS              RESTARTS   AGE
api-6588845f-fjxx7                                             1/1     Running             0          3m3s
bridge-649fd6b75c-6lvnh                                        1/1     Running             0          3m3s
configuration-service-65b4b877cc-4ds5r                         0/1     ContainerCreating   0          45s
eventbroker-go-9c84f5fb-99c8j                                  1/1     Running             0          3m3s
helm-service-78c7565cd8-86bvq                                  1/1     Running             0          3m3s
helm-service-service-create-distributor-869dccd975-6cgjt       1/1     Running             0          3m3s
keptn-nats-cluster-1                                           1/1     Running             0          3m13s
nats-operator-5b9df7cbb7-crwl8                                 1/1     Running             0          3m44s
shipyard-service-75c8bb7d49-zcxr4                              1/1     Running             0          3m3s
shipyard-service-create-project-distributor-955b65d59-jr6nd    1/1     Running             0          3m3s
shipyard-service-delete-project-distributor-54df8745f8-pjvf9   1/1     Running             0          3m3s
[keptn|DEBUG] [2020-02-24 19:38:29] Retry: 19/24 - Wait 10s for pods to start in namespace keptn ...
NAME                                                           READY   STATUS              RESTARTS   AGE
api-6588845f-fjxx7                                             1/1     Running             0          3m14s
bridge-649fd6b75c-6lvnh                                        1/1     Running             0          3m14s
configuration-service-65b4b877cc-4ds5r                         0/1     ContainerCreating   0          56s
eventbroker-go-9c84f5fb-99c8j                                  1/1     Running             0          3m14s
helm-service-78c7565cd8-86bvq                                  1/1     Running             0          3m14s
helm-service-service-create-distributor-869dccd975-6cgjt       1/1     Running             0          3m14s
keptn-nats-cluster-1                                           1/1     Running             0          3m24s
nats-operator-5b9df7cbb7-crwl8                                 1/1     Running             0          3m55s
shipyard-service-75c8bb7d49-zcxr4                              1/1     Running             0          3m14s
shipyard-service-create-project-distributor-955b65d59-jr6nd    1/1     Running             0          3m14s
shipyard-service-delete-project-distributor-54df8745f8-pjvf9   1/1     Running             0          3m14s
[keptn|DEBUG] [2020-02-24 19:38:39] Retry: 20/24 - Wait 10s for pods to start in namespace keptn ...
NAME                                                           READY   STATUS              RESTARTS   AGE
api-6588845f-fjxx7                                             1/1     Running             0          3m24s
bridge-649fd6b75c-6lvnh                                        1/1     Running             0          3m24s
configuration-service-65b4b877cc-4ds5r                         0/1     ContainerCreating   0          66s
eventbroker-go-9c84f5fb-99c8j                                  1/1     Running             0          3m24s
helm-service-78c7565cd8-86bvq                                  1/1     Running             0          3m24s
helm-service-service-create-distributor-869dccd975-6cgjt       1/1     Running             0          3m24s
keptn-nats-cluster-1                                           1/1     Running             0          3m34s
nats-operator-5b9df7cbb7-crwl8                                 1/1     Running             0          4m5s
shipyard-service-75c8bb7d49-zcxr4                              1/1     Running             0          3m24s
shipyard-service-create-project-distributor-955b65d59-jr6nd    1/1     Running             0          3m24s
shipyard-service-delete-project-distributor-54df8745f8-pjvf9   1/1     Running             0          3m24s
[keptn|DEBUG] [2020-02-24 19:38:49] Retry: 21/24 - Wait 10s for pods to start in namespace keptn ...
NAME                                                           READY   STATUS              RESTARTS   AGE
api-6588845f-fjxx7                                             1/1     Running             0          3m34s
bridge-649fd6b75c-6lvnh                                        1/1     Running             0          3m34s
configuration-service-65b4b877cc-4ds5r                         0/1     ContainerCreating   0          76s
eventbroker-go-9c84f5fb-99c8j                                  1/1     Running             0          3m34s
helm-service-78c7565cd8-86bvq                                  1/1     Running             0          3m34s
helm-service-service-create-distributor-869dccd975-6cgjt       1/1     Running             0          3m34s
keptn-nats-cluster-1                                           1/1     Running             0          3m44s
nats-operator-5b9df7cbb7-crwl8                                 1/1     Running             0          4m15s
shipyard-service-75c8bb7d49-zcxr4                              1/1     Running             0          3m34s
shipyard-service-create-project-distributor-955b65d59-jr6nd    1/1     Running             0          3m34s
shipyard-service-delete-project-distributor-54df8745f8-pjvf9   1/1     Running             0          3m34s
[keptn|DEBUG] [2020-02-24 19:38:59] Retry: 22/24 - Wait 10s for pods to start in namespace keptn ...
NAME                                                           READY   STATUS              RESTARTS   AGE
api-6588845f-fjxx7                                             1/1     Running             0          3m44s
bridge-649fd6b75c-6lvnh                                        1/1     Running             0          3m44s
configuration-service-65b4b877cc-4ds5r                         0/1     ContainerCreating   0          86s
eventbroker-go-9c84f5fb-99c8j                                  1/1     Running             0          3m44s
helm-service-78c7565cd8-86bvq                                  1/1     Running             0          3m44s
helm-service-service-create-distributor-869dccd975-6cgjt       1/1     Running             0          3m44s
keptn-nats-cluster-1                                           1/1     Running             0          3m54s
nats-operator-5b9df7cbb7-crwl8                                 1/1     Running             0          4m25s
shipyard-service-75c8bb7d49-zcxr4                              1/1     Running             0          3m44s
shipyard-service-create-project-distributor-955b65d59-jr6nd    1/1     Running             0          3m44s
shipyard-service-delete-project-distributor-54df8745f8-pjvf9   1/1     Running             0          3m44s
[keptn|DEBUG] [2020-02-24 19:39:09] Retry: 23/24 - Wait 10s for pods to start in namespace keptn ...
NAME                                                           READY   STATUS              RESTARTS   AGE
api-6588845f-fjxx7                                             1/1     Running             0          3m54s
bridge-649fd6b75c-6lvnh                                        1/1     Running             0          3m54s
configuration-service-65b4b877cc-4ds5r                         0/1     Terminating         0          96s
configuration-service-65b4b877cc-mbnz8                         0/1     ContainerCreating   0          5s
eventbroker-go-9c84f5fb-99c8j                                  1/1     Running             0          3m54s
helm-service-78c7565cd8-86bvq                                  1/1     Running             0          3m54s
helm-service-service-create-distributor-869dccd975-6cgjt       1/1     Running             0          3m54s
keptn-nats-cluster-1                                           1/1     Running             0          4m4s
nats-operator-5b9df7cbb7-crwl8                                 1/1     Running             0          4m35s
shipyard-service-75c8bb7d49-zcxr4                              1/1     Running             0          3m54s
shipyard-service-create-project-distributor-955b65d59-jr6nd    1/1     Running             0          3m54s
shipyard-service-delete-project-distributor-54df8745f8-pjvf9   1/1     Running             0          3m54s
[keptn|DEBUG] [2020-02-24 19:39:19] Retry: 24/24 - Wait 10s for pods to start in namespace keptn ...
[keptn|ERROR] [2020-02-24 19:39:29] Pods in namespace keptn are not running.
NAME                                     READY   STATUS              RESTARTS   AGE
configuration-service-65b4b877cc-mbnz8   0/1     ContainerCreating   0          16s
[keptn|ERROR] [2020-02-24 19:39:30] Installing Keptn failed.
[keptn|ERROR] [2020-02-24 19:39:30] Stopping keptn installation. Already created resources are not deleted; run keptn uninstall to clean-up.
```
Towards the end of the installation I tried to kill the pod since it might be stuck, but also the new pod could not come up in time.

**Additional context**
For my latest 6 installations on AKS it failed 3 times. So there is a chance that it might run through at the first time.

**Comment by agrimmer** [comment]
This Azure issue is known: https://github.com/kubernetes/kubernetes/issues/69262
Please upgrade to the latest version 1.15.5, 1.15.x, which won't have this issue.
@jetzlstorfer Can you please update the docu accordingly?

**Comment by christian-kreuzberger-dtx** [comment]
As far as I understand the related K8s issue, there is nothing we can fix in Keptn. It's a K8s issue on Azure, and it is fixed in K8s 1.15.5.

As @agrimmer pointed out, let's state this in our docs. I'll also add it to our Known Issues in #1415 for 0.6.1

**Comment by christian-kreuzberger-dtx** [comment]
This has been documented as a known issue and within https://keptn.sh/docs/
